addons/account/tests/test_sequence_mixin.py

```diff
@@ -427,3 +432,193 @@ def test_change_journal_on_first_account_move(self):
         self.assertEqual(move.name, 'MISC/2021/10/0001')
         with Form(move) as move_form:
             move_form.journal_id = journal
+
+    def _create_invoice_form(self, env, date, post=True):
+        with Form(env['account.move'].with_context(default_move_type='out_invoice')) as invoice_form:
+            invoice_form.partner_id = env.ref('base.res_partner_12')
+            invoice_form.invoice_date = date
+            with invoice_form.invoice_line_ids.new() as line_form:
+                line_form.name = 'concurency_test_invoice'
+                line_form.price_unit = 100.0
+                line_form.tax_ids.clear()
+            invoice = invoice_form.save()
+        if post:
+            invoice.action_post()
+        return invoice
+
+    def _create_payment_form(self, env, date):
+        with Form(env['account.payment'].with_context(**{
```

Comment:
What do you think about first saving the context to a variable and then `with_context(**ctx)`?